### PR TITLE
Add Spanner stale read sample.

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

File: spanner/crud.js [hunk 1]
```diff
@@ -163,6 +163,50 @@ function readData (instanceId, databaseId) {
...
+
+  // Uncomment these lines to specify the instance and database to use
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+
+  // Gets a reference to a Cloud Spanner instance and database
+  const instance = spanner.instance(instanceId);
+  const database = instance.database(databaseId);
+
+  // Read rows from the Albums table
```
Comment (ace-n) [inline spanner/crud.js]: Nit: inconsistent comment tense.

File: spanner/crud.js [hunk 2]
```diff
@@ -163,6 +163,50 @@ function readData (instanceId, databaseId) {
...
+
+  const query = {
+    columns: ['SingerId', 'AlbumId', 'AlbumTitle', 'MarketingBudget'],
+    keySet: {
+      all: true
+    }
+  };
+
+  const options = {
+    // Guarantees that all writes that have committed more than 10 seconds ago
```
Comment (ace-n) [inline spanner/crud.js]: Nit: remove `that have` and combine into one line

File: spanner/crud.js [hunk 3]
```diff
@@ -163,6 +163,50 @@ function readData (instanceId, databaseId) {
...
+    exactStaleness: 10
+  };
+
+  albumsTable.read(query, options)
+    .then((results) => {
+      const rows = results[0];
+
+      rows.forEach((row) => {
+        const json = row.toJSON();
+        console.log(`SingerId: ${json.SingerId.value}, AlbumId: ${json.AlbumId.value}, AlbumTitle: ${json.AlbumTitle}, MarketingBudget: ${json.MarketingBudget ? json.MarketingBudget.value : ''}`);
```
Comment (ace-n) [inline spanner/crud.js]: Can we split these into separate lines? This is a lot to read at once...

File: spanner/crud.js [hunk 4]
```diff
@@ -189,10 +233,17 @@ const cli = require(`yargs`)
     {},
     (opts) => readData(opts.instanceName, opts.databaseName)
   )
+  .command(
+    `read-stale <instanceName> <databaseName>`,
+    `Reads data in an example Cloud Spanner table.`,
```
Comment (ace-n) [inline spanner/crud.js]: Nit: mention/explain stale reads in the command description?

e.g. `Reads _stale_ data in an example Cloud Spanner table.`

File: spanner/system-test/spanner.test.js [hunk 5]
```diff
@@ -53,103 +53,139 @@ test.after.always(async (t) => {
 
 // create_database
 test.serial(`should create an example database`, async (t) => {
-  const output = await tools.runAsync(`${schemaCmd} createDatabase "${INSTANCE_ID}" "${DATABASE_ID}"`, cwd);
-  t.true(output.includes(`Waiting for operation on ${DATABASE_ID} to complete...`));
-  t.true(output.includes(`Created database ${DATABASE_ID} on instance ${INSTANCE_ID}.`));
+  const results = await tools.runAsyncWithIO(`${schemaCmd} createDatabase "${INSTANCE_ID}" "${DATABASE_ID}"`, cwd);
```
Comment (ace-n) [inline spanner/system-test/spanner.test.js]: What's the point of this?

`stderr` should be ignored unless we expect an error to occur. (If an error occurs when it shouldn't, our test _should_ fail.)

File: spanner/system-test/spanner.test.js [hunk 6]
```diff
@@ -53,103 +53,139 @@ test.after.always(async (t) => {
...
+  t.regex(output, new RegExp(`Created database ${DATABASE_ID} on instance ${INSTANCE_ID}.`));
 });
 
 // insert_data
 test.serial(`should insert rows into an example table`, async (t) => {
-  let output = await tools.runAsync(`${crudCmd} insert ${INSTANCE_ID} ${DATABASE_ID}`, cwd);
-  t.true(output.includes(`Inserted data.`));
+  const results = await tools.runAsyncWithIO(`${crudCmd} insert ${INSTANCE_ID} ${DATABASE_ID}`, cwd);
+  const output = results.stdout + results.stderr;
+  t.regex(output, new RegExp(`Inserted data.`));
```
Comment (ace-n) [inline spanner/system-test/spanner.test.js]: Replace `t.regex(output, new RegExp($STRING))` with `t.regex(output, /$STRING/)`. Apply this comment throughout your PR.

File: spanner/system-test/spanner.test.js [hunk 7]
```diff
@@ -53,103 +53,139 @@ test.after.always(async (t) => {
...
+  t.regex(output, new RegExp(`Updated data.`));
+});
+
+// read_stale_data
+test.serial(`should read stale data from an example table`, (t) => {
+  t.plan(2);
+  // read-stale-data reads data that is exactly 10 seconds old.  So, make sure
+  // 10 seconds have elapsed since the update_data test.
+  return new Promise((resolve, reject) => {
+    setTimeout(async () => {
```
Comment (ace-n) [inline spanner/system-test/spanner.test.js]: Nit: can you promisify the `setTimeout()` call, and wrap the tests in a `then` as was done [here](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/6fca8cf07d951b9731acd4615310edb8b3e10ef3/pubsub/system-test/topics.test.js#L73-L79)?

File: spanner/crud.js (hunk 1 above)
Comment (jmdobry) [reply spanner/crud.js]: Done

File: spanner/crud.js (hunk 2 above)
Comment (jmdobry) [reply spanner/crud.js]: Done

File: spanner/crud.js (hunk 3 above)
Comment (jmdobry) [reply spanner/crud.js]: Done

File: spanner/crud.js (hunk 4 above)
Comment (jmdobry) [reply spanner/crud.js]: Done.

File: spanner/system-test/spanner.test.js (hunk 5 above)
Comment (jmdobry) [reply spanner/system-test/spanner.test.js]: I've discover lately that sometimes tests/commands fail silently (return a 0 exit code), but _do_ print some useful info to stderr. This improves test debuggability.

File: spanner/system-test/spanner.test.js (hunk 7 above)
Comment (jmdobry) [reply spanner/system-test/spanner.test.js]: Done.

**Comment by jmdobry** [comment]
Comments addressed, merging.

**Pull request merged**